client/app/deskaas/deskaas-change-password/deskaas-change-password.html

```diff
@@ -1,50 +1,41 @@
-<cui-modal>
-    <form novalidate name="form">
-        <cui-modal-header data-on-dismiss="DeskaasGetConsoleAccessCtrl.cancel()"></cui-modal-header>
-        <cui-modal-body data-title=":: 'vdi_pwd_title' | translate">
-            <oui-spinner data-ng-if="vm.flags.init"></oui-spinner>
-            <div data-ng-if="!vm.flags.init">
-                <div>
-                    <ul>
-                        <li data-translate="vdi_pwd_policy_length" data-translate-values="{min : vm.policies.minLength, max : vm.policies.maxLength}"></li>
-                        <li data-translate="vdi_pwd_policy_letterMandatory" data-ng-if="vm.policies.letterMandatory"></li>
-                        <li data-translate="vdi_pwd_policy_lowerMandatory" data-ng-if="vm.policies.lowercaseLetterMandatory"></li>
-                        <li data-translate="vdi_pwd_policy_upperMandatory" data-ng-if="vm.policies.uppercaseLetterMandatory"></li>
-                        <li data-translate="vdi_pwd_policy_digitMandatory" data-ng-if="vm.policies.digitMandatory"></li>
-                        <li data-translate="vdi_pwd_policy_specialMandatory" data-ng-if="vm.policies.specialMandatory"></li>
-                        <li data-translate="vdi_pwd_policy_deniedChars" data-translate-values="{chars : vm.policies.deniedCharsString}"></li>
-                    </ul>
+<form novalidate name="form" class="deskaas-change-password" data-ng-submit="(form.$valid || vm.values.generatePwd) && vm.ok()">
+    <oui-modal data-title="{{ ::'vdi_pwd_title' | translate }}"
+               data-primary-label="{{ ::'common_modal_ok' | translate }}"
+               data-secondary-action="vm.cancel()"
+               data-secondary-label="{{ ::'common_modal_cancel' | translate }}"
+               data-on-dismiss="vm.cancel()"
+               data-loading="vm.flags.init">
+        <div data-ng-if="!vm.flags.init">
+            <ul>
+                <li data-translate="vdi_pwd_policy_length" data-translate-values="{min : vm.policies.minLength, max : vm.policies.maxLength}"></li>
+                <li data-translate="vdi_pwd_policy_letterMandatory" data-ng-if="vm.policies.letterMandatory"></li>
+                <li data-translate="vdi_pwd_policy_lowerMandatory" data-ng-if="vm.policies.lowercaseLetterMandatory"></li>
```

Comment:
Why not use https://docs.angularjs.org/api/ngMessages/directive/ngMessages ?

Reply:
And the scope of this ticket was limited to migrating only the modal. So i will have to pick up the password validation work as a new ticket.